app/experimenter/static/rapid/components/forms/ExperimentForm.tsx

```diff
@@ -0,0 +1,170 @@
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+interface ErrorListProperties {
+  errors: Array<string> | undefined;
+}
+
+const ErrorList: React.FC<ErrorListProperties> = ({ errors }) => {
+  if (!errors) {
+    return null;
+  }
+
+  return (
+    <ul className="text-danger pl-3 my-2">
+      {errors.map((e, i) => (
+        <li key={i}>{e}</li>
+      ))}
+    </ul>
+  );
+};
+
+const ExperimentForm: React.FC = () => {
+  const [formData, setFormData] = React.useState({
+    name: "",
+    objectives: "",
+  });
+  const [errors, setErrors] = React.useState({});
+  const history = useHistory();
+
+  const handleChange = (ev) => {
+    const field = ev.target;
+    setFormData({
+      ...formData,
+      [field.getAttribute("name")]: field.value,
+    });
+  };
+
+  const handleClickSave = async () => {
+    const response = await fetch("/api/v3/experiments/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(formData),
+    });
+    const responseData = await response.json();
+    if (!response.ok) {
+      setErrors(responseData);
+    } else {
+      setErrors({});
+      history.push(`/${responseData.slug}/`);
+    }
+  };
+
+  return (
+    <form>
+      <div className="mb-4">
+        <label className="font-weight-bold" htmlFor="field-name">
+          Public Name
+        </label>
+        <p>
+          Choose a name that describes the experiment you wish to run. This name
+          will be public to users.
+        </p>
+        <input
+          className="form-control w-100"
+          id="field-name"
+          name="name"
+          type="text"
+          value={formData["name"]}
```

Comment:
why not dot notation?

Reply:
whoops! good catch

Reply:
Interesting indeed! I would have expected ESLint to catch this (ref https://eslint.org/docs/rules/dot-notation), but maybe it isn't in a default ESLint config.

Wondering if our ESLint config isn't 100% correct.

```sh
git rev-parse HEAD # bf8b35f3299b2fe1a46ffa4cc0d9177ea4927689
DEBUG=eslint:cli-engine npm run lint

> @experimenter/rapid@1.0.0 lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid
> yarn lint:eslint; yarn lint:tsc

yarn run v1.22.4

$ eslint --no-error-on-unmatched-pattern --color --max-warnings 0 ./*.ts ./*.tsx ./**/*.ts ./**/*.tsx ./**/*.js
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/index.tsx +0ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/components/App.tsx +287ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/__tests__/app.test.js +66ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/__tests__/experimentForm.test.js +198ms
  eslint:cli-engine Linting complete in: 1212ms +73ms
✨  Done in 1.62s.
yarn run v1.22.4

$ tsc --noEmit --project tsconfig.json
✨  Done in 2.68s.
```

What's weird is that it doesn't look like this file is being linted currently. Wondering if the globs in package.json need to be quoted so they're handled by ESLint's engine vs being expanded by my shell.

Because if I quote the globs, I suddenly start seeing some errors:

```
DEBUG=eslint:cli-engine npm run lint

> @experimenter/rapid@1.0.0 lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid
> yarn lint:eslint; yarn lint:tsc

yarn run v1.22.4
$ eslint --no-error-on-unmatched-pattern --color --max-warnings 0 './*.ts' './*.tsx' './**/*.ts' './**/*.tsx' './**/*.js'
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/index.tsx +0ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/components/App.tsx +272ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/components/forms/ExperimentForm.tsx +55ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/components/pages/ExperimentFormPage.tsx +148ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/.eslintrc.js +41ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/__tests__/app.test.js +177ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/__tests__/experimentForm.test.js +46ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/jest-env.setup.js +63ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/jest.config.js +19ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/jest.setup.js +25ms
  eslint:cli-engine Lint /Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/webpack.config.js +33ms
  eslint:cli-engine Linting complete in: 1470ms +40ms

/Volumes/Dev/github/mozilla/experimenter/app/experimenter/static/rapid/components/forms/ExperimentForm.tsx
    8:53  error  'errors' is missing in props validation           react/prop-types
   15:15  error  'errors.map' is missing in props validation       react/prop-types
   70:27  error  ["name"] is better written in dot notation        dot-notation
   73:35  error  ["name"] is better written in dot notation        dot-notation
   89:27  error  ["objectives"] is better written in dot notation  dot-notation
   92:35  error  ["objectives"] is better written in dot notation  dot-notation
  108:35  error  ["feature"] is better written in dot notation     dot-notation
  124:35  error  ["audience"] is better written in dot notation    dot-notation
  140:35  error  ["trigger"] is better written in dot notation     dot-notation
  156:35  error  ["version"] is better written in dot notation     dot-notation

✖ 10 problems (10 errors, 0 warnings)
  8 errors and 0 warnings potentially fixable with the `--fix` option.

error Command failed with exit code 1.
```